lib/philomena_web/controllers/api/json/comment_controller.ex

```diff
@@ -19,14 +22,49 @@ defmodule PhilomenaWeb.Api.Json.CommentController do
         |> put_status(:not_found)
         |> text("")
 
-      comment.image.hidden_from_users ->
+      true ->
+        json(conn, %{comment: CommentJson.as_json(comment)})
+    end
+  end
+  def show(conn, %{"id" => id}) do
+    comment = 
+      Comment
+      |> join(:inner, [c], _ in assoc(c, :image))
+      |> where(id: ^id)
+      |> where(destroyed_content: false)
+      |> where([_c, i], i.hidden_from_users == false)
+      |> preload([:image, :user])
+      |> Repo.one()
+
+    cond do
+      is_nil(comment) ->
         conn
-        |> put_status(:forbidden)
+        |> put_status(:not_found)
         |> text("")
 
       true ->
         json(conn, %{comment: CommentJson.as_json(comment)})
+    end
+  end
+
+  def index(conn, %{"image_id" => image_id}) do
+    page = conn.assigns.pagination.page_number
+    comments = 
+      Comment
+      |> join(:inner, [c], _ in assoc(c, :image))
+      |> where(destroyed_content: false)
+      |> where([_c, i], i.hidden_from_users == false and i.id == ^image_id)
+      |> preload([:image, :user])
+      |> Repo.all()
+
+    case comments do
+      [] ->
```

Comment:
This isn't the appropriate response (imo, doesn't make any sense to 404 the comments section on an image with no comments)